provisioning/roles/environment/tasks/nifi.yml

```diff
@@ -0,0 +1,174 @@
+---
+- name: Add nifi group
+  become: true
+  group:
+    name: nifi
+    state: present
+
+- name: Add nifi user
+  become: true
+  user:
+    name: nifi
+    group: nifi
+    createhome: true
+
+- name: Create folder for latest version of NiFi
+  file:
+    path: /opt/nifi
+    state: directory
+    owner: nifi
+    group: nifi
+    mode: 0755
+  become: true
+
+- name: Create Folder for NiFi
+  file:
+    path: /opt/nifi/
+    state: directory
+    owner: nifi
+    group: nifi
+    mode: 0755
+    recurse: true
+  become: true
+
+# If online, will download from the url then unzip
+- name: Unzip Version {{ nifi_version }} of NiFi
+  unarchive:
+    src: "{{ nifi.url if not is_offline else nifi.path }}"
+    dest: /opt/nifi
+    mode: 0755
+    remote_src: "{{ true if not is_offline else false | bool }}"
+  become: true
+
+# If online, will download from the url then unzip
+- name: Unzip NiFi Tools
+  unarchive:
+    src: "{{ nifi_tools.url if not is_offline else nifi_tools.path }}"
+    dest: /opt/nifi
+    mode: 0755
+    remote_src: "{{ true if not is_offline else false | bool }}"
+  become: true
+
```

Comment:
The provision fails here for me on Windows 11.  This is the error message I got today. 
<img width="866" alt="image" src="https://github.com/ctc-oss/noe-setup/assets/72815523/c27a3a71-359b-4148-9059-e6b61beef025">
I sent a teams message Friday with a different message, but it looks like it may not have sent properly.  I don't remember exactly, but I believe that message had something to do with an undefined user.

Reply:
@rthomas320 This looks like an error with unzipping the nifi-toolkit zip file, not a user error. Nothing in that command uses something dealing with a user. Since you were successfully able to unzip the NiFi zip but not the nifi-toolkit zip, it should be something with the nifi-toolkit zip file. Are you sure it was downloaded properly?

Reply:
I tried running the provisioning Friday, I received an error about an undefine ansible user.  I captured a snippet of the error message and sent it in teams, but it appears it did not send successfully.  I was able to see the failed message when I logged in the next day when I opened teams but all teams functionality was greyed out. When I closed and re-opened teams I received a message to authenticate and change my password.  Without my company laptop, I am unable to change my password.  I don't see a separate command to download the nifi toolkit file.  The only path I see is in the unarchive command.  No zip file was in that directory.  In a previous version of the original provisioning, I was using the unarchive command, but I couldn't get it to work with Windows WSL as the host OS  and Rocky9 as the VM.  In this instance with the PR code I was trying to bring up a Rocky8 VM.  I can try the provisioning again, but I will destroy the VM and start over.

Reply:
I can't get the provisioning past the nifi unzip.  I'm not going to do any more until I am back to work on the 10th.

Reply:
@rthomas320 Does the nifi toolkit zip exist on your machine where the nifi zip is? If so can you try opening and/or unzipping on your host? It might be that the file didn't download properly when you tried to download files for an offline deployment. Or are you not trying to do an offline deployment?

Reply:
@rthomas320 were you able to get any further?